api/src/modules/Session/model.ts

```diff
@@ -74,6 +82,51 @@ const generateTokenAndSession = async (
   };
 };
 
+async function createUserIfNotExists(prisma: PrismaClient, wallet_address: string) {
+  const user = await prisma.user.findUnique({
+    where: {
+      id: wallet_address,
+    }
+  })
+  if (!!user) {
+    return user
+  }
+  return await prisma.user.create({
+    data: {
+      id: wallet_address,
+    }
+  })
+
+}
+async function createPendingSession(prisma: PrismaClient): Promise<string> {
```

Comment:
- Please improve consistency by calling it the same way
- Please improve consistency by returning the same type as the name suggests
```suggestion
async function createChallenge(prisma: PrismaClient): Promise<Challenge> {
```